src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -33,6 +34,41 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public class ExceptionHelper
     {
+#if !NET_4_5 && !PORTABLE
+        private static readonly Action<Exception> PreserveStackTrace;
+
+        static ExceptionHelper()
+        {
+            var method = typeof(Exception).GetMethod("InternalPreserveStackTrace", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (method != null)
+            {
+                try
+                {
+                    PreserveStackTrace = (Action<Exception>)Delegate.CreateDelegate(typeof(Action<Exception>), method);
+                    return;
+                }
+                catch (InvalidOperationException) { }
+            }
+            PreserveStackTrace = _ => { };
+        }
+#endif
+
+        /// <summary>
+        /// Rethrows an exception, preserving its stack trace
+        /// </summary>
+        /// <param name="exception">The exception to rethrow</param>
+        public static void Rethrow(Exception exception)
```

Comment:
We have made framework internals visible to the tests, so you should be able to make these internal. If not, then we should look at why. I would like to get to a point where we are exposing fewer and fewer internal classes as public to prevent people from using them incorrectly or even just depending on classes that may change.